src/main/kotlin/com/atlassian/performance/tools/jiraactions/api/page/IssueNavigatorPage.kt

```diff
@@ -6,40 +6,63 @@ import com.atlassian.performance.seleniumjs.NativeExpectedConditions.Companion.p
 import com.atlassian.performance.tools.jiraactions.api.webdriver.JavaScriptUtils
 import org.openqa.selenium.By
 import org.openqa.selenium.WebDriver
+import org.openqa.selenium.support.ui.ExpectedConditions.elementToBeClickable
 import java.time.Duration
 
 open class IssueNavigatorPage(
     private val driver: WebDriver,
     val jql: String
 ) {
+    private val detailView = and(
+        or(
+            presenceOfElementLocated(By.cssSelector("ol.issue-list")),
+            presenceOfElementLocated(By.id("issuetable")),
+            presenceOfElementLocated(By.id("issue-content"))
+        ),
+        presenceOfElementLocated(By.id("key-val")),
+        presenceOfElementLocated(By.className("issue-body-content"))
+    )
+    private val listView = and(
+        presenceOfElementLocated(By.id("issuetable")),
+        presenceOfElementLocated(By.cssSelector(".issuerow.focused"))
+    )
+    private val emptyResults = presenceOfElementLocated(By.className("no-results-hint"))
 
-    /**
-     * Requires an issue detail pane to be visible.
-     * For example, if there are no issues found, this will time out.
-     */
     fun waitForIssueNavigator(): IssueNavigatorPage {
         val jiraErrors = JiraErrors(driver)
         driver.wait(
-            Duration.ofSeconds(30),
+            Duration.ofSeconds(10),
             or(
-                and(
-                    or(
-                        presenceOfElementLocated(By.cssSelector("ol.issue-list")),
-                        presenceOfElementLocated(By.id("issuetable")),
-                        presenceOfElementLocated(By.id("issue-content"))
-                    ),
-                    presenceOfElementLocated(By.id("key-val")),
-                    presenceOfElementLocated(By.className("issue-body-content"))
-                ),
-                presenceOfElementLocated(By.className("no-results-hint")),
+                detailView,
+                listView,
+                emptyResults,
                 jiraErrors.anyCommonErrorNative()
             )
         )
         return this
     }
 
+    fun switchLayoutTo(view: IssueNavigatorView): IssueNavigatorPage {
+        val currentLayout = getCurrentLayout()
+        if (currentLayout != view) {
+            driver.wait(elementToBeClickable(By.id("layout-switcher-button"))).click()
+            driver.wait(elementToBeClickable(By.cssSelector("a[data-layout-key=${view.selector}]"))).click()
+        }
+        return this
+    }
+
+    private fun getCurrentLayout(): IssueNavigatorView {
+        driver.wait(Duration.ofSeconds(10), presenceOfElementLocated(By.className("results-panel")))
```

Comment:
Unfortunately, to detect the currently selected view we need to wait for search results to appear.

Reply:
Hmm, we're using the sequence of `switchLayoutTo` and then `waitForIssueNavigator`. So if `By.className("results-panel")` is earlier than the `waitForIssueNavigator` conditions, it will not affect the measurements. In other words, if `waitForIssueNavigator` was already waiting for `results-panel` (indirectly), then wer're good.

If not, then we could do the switch *after* the page load. It would distort the proportions of the layouts, but it would preserve the measurement, which is more important.

Reply:
So I checked on dogfooding and indeed `results-panel` is a container for all the selectors in `val detailView` and `val listView`. So we're good.

But, the sequence is  `goToIssueNavigator(jqlQuery).switchLayoutTo(view).waitForIssueNavigator()`. So the `goToIssueNavigator` already performs the search, then `switchLayoutTo` can interrupt it, switch layout and re-trigger the search. It means that sometimes we'll make 1,5 search, affecting measurements.

So we can:
* bite the bullet and count on a low switch count
* switch after search
* prevent early search (no `jql=` in URI)

The `goToIssueNavigator` triggers JQL, because historically there was no other way. Going to "search for issues" was sending an empty JQL, which was interpreted as "all issues".
![image](https://user-images.githubusercontent.com/1194704/227201248-24374cda-689b-4bd1-ad7a-8a532b4f0733.png)
Since then, Jira changed empty JQL to mean "no search". So technically we can now `goToIssueNavigator` without a JQL, switch layout and then search.

Reply:
I chose this order of operations because, as you have noticed, to detect the currently selected layout, we need the `results-panel` to be present. If we reverse the order, there are some questions that arise: 
Should changing the layout be included in the measurement? If so, shouldn't we repeat the verification which is currently being done in the `waitForIssueNavigator` function at the end? If not, do we even need to change the layout at all? 
I must admit that I didn't consider the option of separating the search from actually going to the Issue Search view, perhaps that option is the best...

Reply:
Switching views should be measured separately from searching issues. We can optimistically assume that switches won't be that frequent, if the scenario doesn't mix both views in one action list. But even then we can expect a lot of switches when the same Jira user is reused across VUs.

So the optimistic assumption must come with a verification - counting the switches. I'll just make a nested action for the switching part. It will have to inform us about each switch attempt and whether it actually switched or just detected that the view is already there. I'll do that in my PR->PR (#118).

Reply:
Done in https://github.com/atlassian/jira-actions/pull/118/commits/0acb5fc123efd40c4e1b270e40e56958c8c1f9dc